extensions/notebook/src/jupyter/jupyterKernel.ts

```diff
@@ -63,6 +63,12 @@ export class JupyterKernel implements nb.IKernel {
 		return true;
 	}
 
+	public get requiresConnection(): boolean {
+		// TODO would be good to have a smarter way to do this.
+		// for now only Spark kernels need a connection
+		return !!(this.kernelImpl.name && this.kernelImpl.name.toLowerCase().indexOf('spark') > -1);
```

Comment:
SQL also needs connection.

Reply:
Yes, that's handled in the SQLSession - see separate file for that. This only covers Jupyter-provided kernels.